### DISCUSSION TITLE
cost model generator output(relu1) size is inconsistent

### DISCUSSION THREAD

**Discussion by kohankhaki** [halide/Halide#6263] [Q&A]
In cost_model_generator.cpp, relu1 is the output of the network and its size should be (27, w, n) by definition (w is the number of stages and n is the number of batches). But in line 383, relu1(i, w, n) is being used i is in range of (0, 32). This is inconsistent to the size of the relu1. 
Also, there is another questions. In line 383, cost is added to 0.0f * relu1(i, w, n) which is zero. What is the point of this?

**Comment by abadams** [comment]
For performance, it's much simpler to compute 32 activations than 27, so we do that, and let 5 go unused. This way we can add more features without having to reschedule the code. Line 383 is adding a false dependence on all 32 so that Halide thinks they are all used.
